### PR TITLE
Fixed error with models when globe is undefined

### LINKED ISSUE #5638: Models broken when Globe is disabled
The below code snippet fails because globe is false, removing the line fixes the example.  This is a regression but I'm not sure which version is stopped working in.  (Note you have to remove the line, not set it to true because of the way Viewer options work.)

```javascript
var viewer = new Cesium.Viewer('cesiumContainer', {
    globe: false
});


var position = Cesium.Cartesian3.fromDegrees(-123.0744619, 44.0503706, 5000);
var heading = Cesium.Math.toRadians(135);
var pitch = 0;
var roll = 0;
var hpr = new Cesium.HeadingPitchRoll(heading, pitch, roll);
var orientation = Cesium.Transforms.headingPitchRollQuaternion(position, hpr);

var entity = viewer.entities.add({
    position : position,
    orientation : orientation,
    model : {
        uri : '../../SampleData/models/CesiumAir/Cesium_Air.glb',
        minimumPixelSize : 128,
        maximumScale : 20000
    }
});
viewer.trackedEntity = entity;
```

Here's the callstack

```
TypeError: Cannot read property 'terrainProviderChanged' of undefined
    at Scene.get (http://cesiumjs.org/Cesium/Source/Scene/Scene.js:919:34)
    at new Model (http://cesiumjs.org/Cesium/Source/Scene/Model.js:532:18)
    at Function.Model.fromGltf (http://cesiumjs.org/Cesium/Source/Scene/Model.js:1203:21)
    at ModelVisualizer.update (http://cesiumjs.org/Cesium/Source/DataSources/ModelVisualizer.js:121:31)
    at DataSourceDisplay.update (http://cesiumjs.org/Cesium/Source/DataSources/DataSourceDisplay.js:262:37)
    at Viewer._onTick (http://cesiumjs.org/Cesium/Source/Widgets/Viewer/Viewer.js:1572:49)
    at Event.raiseEvent (http://cesiumjs.org/Cesium/Source/Core/Event.js:142:30)
    at Clock.tick (http://cesiumjs.org/Cesium/Source/Core/Clock.js:308:21)
    at CesiumWidget.render (http://cesiumjs.org/Cesium/Source/Widgets/CesiumWidget/CesiumWidget.js:685:43)
    at render (http://cesiumjs.org/Cesium/Source/Widgets/CesiumWidget/CesiumWidget.js:71:32)
```

This is probably multiple trivial bugs.  Scene should not access the globe unless it's defined and Model should not subscribe to the terrainChanged event if it's undefined.

### PR DESCRIPTION
**Pull request by ggetz**
Fixes #5638

Has `Scene.js` check if `globe` is defined before accessing it properties, and also has `Model.js` check if the `terrainProviderChanged` event is defined before subscribing.

### CODE REVIEW

**Comment by ggetz** [comment]
@mramato Can you review when you get a chance? (I can't assign reviewers directly)

**Comment by hpinkos** [comment]
This looks good to me @ggetz!  It looks like `BillboardCollection` needs the same change, want to do that as well?

**Comment by ggetz** [comment]
@hpinkos Sure

**Comment by ggetz** [comment]
@hpinkos Updated!

**Comment by hpinkos** [comment]
@ggetz merge in master

**Comment by ggetz** [comment]
@hpinkos Merged!

File: Source/Scene/Model.js [hunk 1]
```diff
@@ -528,7 +528,7 @@ define([
         this._removeUpdateHeightCallback = undefined;
         var scene = options.scene;
         this._scene = scene;
-        if (defined(scene)) {
+        if (defined(scene) && defined(scene.terrainProviderChanged)) {
             scene.terrainProviderChanged.addEventListener(function() {
```
Comment (hpinkos) [inline Source/Scene/Model.js]: This event listener needs to be removed in `Model.prototype.destroy`

**Comment by hpinkos** [comment]
And add specs. And after that I think this will be ready =)

**Comment by ggetz** [comment]
Thanks @hpinkos, updated!

**Comment by ggetz** [comment]
@hpinkos Could you take another look at this when you get the chance?

File: CHANGES.md [hunk 2]
```diff
@@ -40,6 +41,8 @@ Change Log
 * Updated `Billboard`, `Label` and `PointPrimitive` constructors to clone `NearFarScale` parameters [#5654](https://github.com/AnalyticalGraphicsInc/cesium/pull/5654)
 * Added `FrustumGeometry` and `FrustumOutlineGeometry`. [#5649](https://github.com/AnalyticalGraphicsInc/cesium/pull/5649)
 * Added an `options` parameter to the constructors of `PerspectiveFrustum`, `PerspectiveOffCenterFrustum`, `OrthographicFrustum`, and `OrthographicOffCenterFrustum` to set properties. [#5649](https://github.com/AnalyticalGraphicsInc/cesium/pull/5649)
+* Added `ClassificationPrimitive` which defines a volume and draws the intersection of the volume and terrain or 3D Tiles. [#5625](https://github.com/AnalyticalGraphicsInc/cesium/pull/5625)
+* Fix for dynamic polylines with polyline dash material [#5681](https://github.com/AnalyticalGraphicsInc/cesium/pull/5681)
```
Comment (hpinkos) [inline CHANGES.md]: Remove these

**Comment by ggetz** [comment]
Fixed!

**Comment by hpinkos** [comment]
Thanks @ggetz !

**Pull request merged**